lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -1358,30 +1360,21 @@ namespace Js
 #endif
 
         const char16* pThisStr = pThis->GetString();
-        int thisStrCount = pThis->GetLength();
-
         const char16* pThatStr = pThat->GetString();
-        int thatStrCount = pThat->GetLength();
+        
+        int result = UnicodeText::LogicalStringCompare(pThisStr, pThatStr);
```

Comment:
Strings can have embedded nulls in JS, so we will need to change this API to pass in the actual length of each string, rather than assuming null termination (especially since GetString is not required to actually give back a null-terminated buffer -- that's what GetSz is for). Since we obviously don't have a test for this case, it would be great to add one!